workflow/rules/peakcall.smk

```diff
@@ -286,4 +286,40 @@ rule SICER:
             mv ${{tmp}}/{params.name}-W300-G600.scoreisland {output.bed}
         fi
     fi
+    """
+
+rule MEME:
+    input:
+        bed = lambda w: join(workpath, w.PeakTool, w.name, w.name + PeakExtensions[w.PeakTool])
+    output:
+        meme_out = join(workpath, "MEME", "{PeakTool}", "{name}_meme", "meme-chip.html"),
+        ame_out = join(workpath, "MEME", "{PeakTool}", "{name}_ame", "ame.html")
+    params:
+        rname='MEME',
+        ref_fa=config['references'][genome]['GENOME'],
+        meme_vertebrates_db=config['references'][genome]['MEME_VERTEBRATES_DB'],
+        meme_euk_db=config['references'][genome]['MEME_EUKARYOTE_DB'],
+        meme_genome_db=config['references'][genome]['MEME_GENOME_DB'],
+        oc=join(workpath, "MEME", "{PeakTool}", "{name}"),
+        tmpdir=tmpdir,
+        outfa="{name}.fa",
+        ntasks=int(28)
+    shell: """
+    module load meme
+    module load bedtools
+    if [ ! -d "{params.tmpdir}" ]; then mkdir -p "{params.tmpdir}"; fi
+    tmp=$(mktemp -d -p "{params.tmpdir}")
+    trap 'rm -rf "${{tmp}}"' EXIT
+
+    bedtools getfasta -fi {params.ref_fa} -bed {input.bed} -fo ${{tmp}}/{params.outfa}
+    meme-chip \\
+      --oc {params.oc}_meme \\
+      -db {params.meme_vertebrates_db} \\
```

Comment:
@asyakhl Will this work if a user selects rheMac10 as the genome? For this reference genome, it is set to an empty string. I am not sure if that will cause an error when it runs:
![image](https://github.com/OpenOmics/chrom-seek/assets/18038345/80c45f1e-ecea-44b7-be47-be2384740a23)

Reply:
If it does cause a problem, can we add a lambda statement to build the `-db DB` string within the params section of the snakemake rule?